### ISSUE TITLE
IndexLink may not active correctly?

### ISSUE THREAD

**Issue by leozdgao** [rackt/react-router#2310]
Here is my route:

```jsx
<Route>
  <Route path="/" component={App}>
    <IndexRoute component={Dashboard} />
    <Route path="task" component={Task} />
    <Route path="team" component={Team} />
  </Route>
  <Route path="*" component={Page404} />
</Route>
```

I put my Page404 out, because I don't want it share components with App.

And my navbar like this:

```jsx
<Nav>
  <IndexLink to='/' activeClassName='active' />
  <Link to='/task' activeClassName="active" />
  <Link to='/team' activeClassName="active" />
</Nav>
```

If I use IndexLink, it never get 'active' class, if I use Link, it always 'active'. (Should not use Link here I'm sure).

But this route works good:
```jsx
<Route path="/" component={App}>
    <IndexRoute component={Dashboard} />
    <Route path="task" component={Task} />
    <Route path="team" component={Team} />
</Route>
```

I just start to use 1.0.0, did I miss something?

**Comment by taion** [comment]
This was fixed by #2237. This is something like the third time you've tried to report this same issue. Stop it.

**Comment by leozdgao** [comment]
@taion Do not understand about "the third time", maybe because I have edited it 2 times to make my description and sample code more clear. I'm sorry if it disturbed you. I may check old issues first next time. Thx.
